### PR TITLE
LPS-119636 Use useStateSafe in Content Performance

### PR DESCRIPTION
**Pull request by dgarciasarai**
**Description**

Since [this PR](https://github.com/brianchandotcom/liferay-portal/pull/91831), we can use `useStateSafe` from `frontend-js-web` in our components so I changed it in Content Performance (in those that I think applied).

### CODE REVIEW

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TotalCount.js [hunk 1]
```diff
@@ -42,20 +40,13 @@ function TotalCount({
 	useEffect(() => {
 		if (validAnalyticsConnection) {
 			dataProvider()
-				.then((value) => {
-					if (isMounted()) {
-						setValue(value);
-					}
-				})
+				.then((value) => setValue(value))
```
Comment (wincent) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TotalCount.js:43]: Just golfing, but you could make this:

```js
.then(setValue)
```

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TrafficSources.js [hunk 2]
```diff
@@ -55,26 +55,18 @@ export default function TrafficSources({
...
 
 	useEffect(() => {
 		if (validAnalyticsConnection) {
 			dataProvider()
-				.then((trafficSources) => {
-					if (isMounted()) {
-						setTrafficSources(trafficSources);
-					}
-				})
+				.then((trafficSources) => setTrafficSources(trafficSources))
```
Comment (wincent) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TrafficSources.js:63]: Same thing here; could be:

```js
.then(setTrafficSources)
```

**Review by wincent — APPROVED** [review]

**Comment by dgarciasarai** [comment]
ci:stop

**Comment by dgarciasarai** [comment]
Thanks @wincent, I'll change it, it's more elegant :)

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
Hey @liferay-frontend, can we forward this? Thanks 😄

**Comment by wincent** [comment]
ci:forward

**Comment by wincent** [comment]
> Hey @liferay-frontend, can we forward this? Thanks 😄

I just hit `ci:forward` on it — seeing as I already approved it earlier, I think it's fine for you to forward it yourself given that you only made a tiny (and suggested) change on top of what I had reviewed.

**Comment by dgarciasarai** [comment]
Thanks @wincent!

**Pull request closed without merging**